static/app/views/automations/components/automationBuilder.tsx

```diff
@@ -0,0 +1,250 @@
+                    }),
+                  }}
+                  inline={false}
+                  isSearchable={false}
+                  isClearable={false}
+                  name="triggers.logicType"
+                  value={state.triggers.logicType}
+                  onChange={value =>
+                    dispatch({type: 'UPDATE_WHEN_LOGIC_TYPE', logicType: value})
+                  }
+                  required
+                  flexibleControlStateSize
+                  options={TRIGGER_MATCH_OPTIONS}
+                  size="xs"
+                />
+              </EmbeddedWrapper>
+            ),
+          })}
+        </StepLead>
+      </Step>
+      <RuleNodeList
+        placeholder={t('Select a trigger...')}
+        conditions={state.triggers.conditions}
+        group="triggers"
+        onAddRow={type => dispatch({type: 'ADD_WHEN_CONDITION', conditionType: type})}
+        onDeleteRow={index => dispatch({type: 'REMOVE_WHEN_CONDITION', index})}
+        updateCondition={(index, comparison) =>
+          dispatch({type: 'UPDATE_CONDITION', index, comparison})
+        }
+      />
+
+      {state.actionFilters.map((_, index) => (
+        <ActionFilterBlock
+          key={index}
+          id={index}
+          onDelete={() => dispatch({type: 'REMOVE_IF', groupIndex: index})}
+          addIfCondition={(groupIndex, type) =>
+            dispatch({type: 'ADD_IF_CONDITION', groupIndex, conditionType: type})
+          }
+          removeIfCondition={(groupIndex, conditionIndex) =>
+            dispatch({type: 'REMOVE_IF_CONDITION', groupIndex, conditionIndex})
+          }
+          updateIfCondition={(groupIndex, conditionIndex, comparison) =>
+            dispatch({
+              type: 'UPDATE_IF_CONDITION',
+              groupIndex,
+              conditionIndex,
+              comparison,
+            })
+          }
+          updateIfLogicType={(groupIndex, logicType) =>
+            dispatch({type: 'UPDATE_IF_LOGIC_TYPE', groupIndex, logicType})
+          }
+        />
+      ))}
+      <span>
+        <PurpleTextButton
+          borderless
+          icon={<IconAdd />}
+          size="xs"
+          onClick={() => dispatch({type: 'ADD_IF'})}
+        >
+          {t('If/Then Block')}
+        </PurpleTextButton>
+      </span>
+      <span>
+        <Button icon={<IconMail />}>{t('Send Test Notification')}</Button>
+      </span>
+    </Flex>
+  );
+}
+
+interface ActionFilterBlockProps {
+  addIfCondition: (groupIndex: number, type: string) => void;
+  id: number;
+  onDelete: () => void;
+  removeIfCondition: (groupIndex: number, conditionIndex: number) => void;
+  updateIfCondition: (
+    groupIndex: number,
+    conditionIndex: number,
+    comparison: any
+  ) => void;
+  updateIfLogicType: (groupIndex: number, logicType: DataConditionGroupLogicType) => void;
+}
+
+function ActionFilterBlock({
+  id,
+  onDelete,
+  addIfCondition,
+  removeIfCondition,
+  updateIfCondition,
+  updateIfLogicType,
```

Comment:
Any benefit to passing these as props vs passing the indices down and calling `dispatch` directly in this component? Either way is fine, just curious if you considered other patterns.

Reply:
My original intent was to only give the `ActionFilterBlock` the ability to update action filter details, which is why I didn't pass `dispatch` directly. Looking at it now it would probably be cleaner to just call `dispatch` directly